### ISSUE TITLE
Some ideas

### ISSUE THREAD

**Issue by PGTBoos** [marcusbuffett/command-line-chess#49]
Was looking at the code i'm on a jupyter notebook though.
Could it be converted to it? , 
I've still not cracked the looking ahead for moves..

Though for your code.
- Maybe an idea, what if all the potential moves was just selected from     list(board.legal_moves)  ?
- Bishop/Knight values, how about adjusting it during the game a single bishop is less then a pair
  its not just comparing number off bishops vs knights, their value change slightly. (i'd even prefer 2 knights end games).
- Not sure if you allready do this
   If you check several moves ahead, one doesnt need to recalculate if a specific move is indeed played, top results can be cashed
   and if so one can expand on previous results, which might even then get a bonus 1 level deeper.
- Not sure if you take it in the calculation, diagonal pawn walls, might get a higher peace value.

**Comment by ClasherKasten** [comment]
> Could it be converted to it? ,

No, I don't see a good, practical reason converting this project to a Jupyter notebook. Is there a good reason why you would like to have it converted to a Jupyter notebook?

> Maybe an idea, what if all the potential moves was just selected from list(board.legal_moves)

I noticed that `board` has no attribute `legal_moves`. Could you please point to a file and line number where you can find that? Or is this an intentionally introduced new attribute? If so, please describe how it should look and behave.

> Bishop/Knight values, how about adjusting it during the game a single bishop is less then a pair
its not just comparing number off bishops vs knights, their value change slightly. (i'd even prefer 2 knights end games).

This is a good idea, if you wanna do it and have time for it, you can open a PR.

> Not sure if you allready do this
If you check several moves ahead, one doesnt need to recalculate if a specific move is indeed played, top results can be cashed
and if so one can expand on previous results, which might even then get a bonus 1 level deeper.

I didn't see this kind of behaviour anywhere in the code. It sounds like a good idea too, but I guess the problem will be that right now it wont change much, as you can read in the `README.md` under **Technical stuff** it's a very basic and slow AI. So my apprehension is that it needs to much time for often recalculations when the AI wrong predicts the best moves and with a too low lookahead cache the AI can recalculate things fast enough that we don't need it (As seen on the screenshots, 2 moves lookahead is ok to calculate but 3 and above is very time consuming and eventually for this it could make sense to use a cache. But if we choose a number higher than 3, players would wait too long for recalculations).

> Not sure if you take it in the calculation, diagonal pawn walls, might get a higher peace value.

This is again a very interesting idea. if you wanna do it and have time for it, you can open a PR.

**Comment by PGTBoos** [comment]
What is a PR ?
For me it be nice on jupyter since all neural network lives there at my pc.
Might attract other Devs too if published like that.


Outlook voor Android downloaden<https://aka.ms/ghei36>
________________________________

**Comment by PGTBoos** [comment]
Oh and about legal moves this is how I do it at my current chess jupyter notebook.

!pip install chess
import chess

board = chess.Board()
board.turn == chess.WHITE
print(str(board))
list(board.legal_moves)
move = list(board.legal_moves)[0]  # take the first move  (with no score calucalation at all)
print(move.uci())

**Comment by ClasherKasten** [comment]
`!pip install chess` is wrong.
The package name is `cl-chess`, so the correct command would be `!pip install cl-chess`.

**Comment by ericborgos** [comment]
After some experimentation, I got it to work in Google Colab. Here's the code:
!git clone https://github.com/marcusbuffett/command-line-chess
%cd /content/command-line-chess
!pip install .
!chess

**Comment by ClasherKasten** [comment]
For only installing it, and not needing the code, cloning isn't required, because `pip` can install from git repositories. See documentation for this [here](https://pip.pypa.io/en/stable/topics/vcs-support/)
